### ISSUE TITLE
add configuration option for Proxy Protocol

### ISSUE THREAD

**Issue by devmage** [dropwizard/dropwizard#1176]
Dropwizard currently provides a configuration option to read `X-Forwarded-*` headers, `useForwardedHeaders`, but provides no similar configuration option to read [HAProxy PROXY protocol](http://www.haproxy.org/download/1.5/doc/proxy-protocol.txt) data frames, such as those able to be sent through by [AWS ELBs](http://docs.aws.amazon.com/ElasticLoadBalancing/latest/DeveloperGuide/enable-proxy-protocol.html). The NGINX blog has [a write-up](https://www.nginx.com/blog/realtime-applications-nginx/) about this as well.

In vanilla Jetty, this is accomplished using the [ProxyConnectionFactory](http://download.eclipse.org/jetty/stable-9/apidocs/org/eclipse/jetty/server/ProxyConnectionFactory.html), as described in [Jetty's docs on configuring Connectors](http://www.eclipse.org/jetty/documentation/current/configuring-connectors.html#d0e4404).

Dropwizard should include a configuration option to support reading the PROXY protocol.

**Comment by devmage** [comment]
After some digging, this appears to be a Jetty 9.3.x feature, so this would entail a Jetty version bump: https://github.com/eclipse/jetty.project/commit/a308c087edae79917b40dbc8e1b5f0c0b3b15b12

**Comment by arteam** [comment]
#1036 is now resolved and Dropwizard HEAD supports Jetty 9.3.*

**Comment by devmage** [comment]
@arteam thanks for the tip! I'll make sure my colleagues are aware.

**Comment by jplock** [comment]
Do we still need to introduce a configuration option for this or does Jetty 9.3.x fix this?

**Comment by nickbabcock** [comment]
Hopefully you're able to do what you want with Jetty 9.3 in Dropwizard. Let us know if not.

**Comment by wedgwoodtom** [comment]
I think this is still open.  I have the same issue and see no way to configure tcp proxy protocol support in dropwizard although Jetty supports it via a startup module.

**Comment by nickbabcock** [comment]
Ok I will re-open. Ideally one might get away with doing:

```java
((ServerConnector) connector).addFirstConnectionFactory(new ProxyConnectionFactory())
```

But in the `Application::run` the connectors are not yet built so there doesn't appear to be a way to interject. So it looks like there may need to be some level of dropwizard configuration to allow the proxy protocol.

**Comment by tyagiakhilesh** [comment]
I tried the suggested solution but it doesn't work! If I do as stated above, then exactly at what point does this break, but with my haproxy setup, I see 
```
Bad Message 400

reason: Illegal character 0x0
```
I tested this with pure jetty. The above error happens when server side is not able to handle proxy protocol header.

**Comment by tyagiakhilesh** [comment]
Hello Guys, I have this very simple change. And It does achieve what we want to do here. But I need some help this make this change configurable. Can someone help me on two counts:
1. How to make this configurable (probably I shall be able to figure out myself if I spent enough time in code; but any head start/hints are greatly appreciated)
2. Is this the right place to insert the code ? Or I did grossly missed it ? 

I have tested this change. And it works in simplest of the case. I have tested only application port and only http. But looking at code, https should work fine as well.
